src/main/java/org/reaktivity/nukleus/maven/plugin/internal/ast/parse/AstParser.java

```diff
@@ -117,13 +117,12 @@ public AstScopeNode visitScope(
 
         if (scopeBuilders.peekLast() != null)
         {
-            String qualifiedName = new StringBuilder().append(scopeNameMap.get(scopeBuilders.peekLast().name()))
-                .append("::").append(name).toString();
-            scopeNameMap.put(name, qualifiedName);
+            String qualifiedName = String.format("%s::%s", qualifiedNamesByLocalName.get(scopeBuilders.peekLast().name()), name);
+            qualifiedNamesByLocalName.put(name, qualifiedName);
         }
         else
         {
-            scopeNameMap.put(name, name);
+            qualifiedNamesByLocalName.put(name, name);
         }
```

Comment:
Suggest the following change.
```
String qualifiedName = name;
AstScopeNode.Builder scopeBuilder = scopeBuilders.peekLast();
if (scopeBuilder != null)
{
    String scopeName = scopeBuilder.name();
    String qualifiedScopeName = qualifiedNamesByLocalName.get(scopeName);
    qualifiedName = String.format("%s::%s", qualifiedScopeName, name);
}
qualifiedNamesByLocalName.put(name, qualifiedName);
```

Reply:
Also, if this code is same as `putQualifiedName`, suggest moving it there and calling `putQualifiedName` here instead.
Consider renaming method `putQualifiedName` as `visitLocalName`.